substrate/frame/support/procedural/src/runtime/parse/mod.rs

```diff
@@ -187,6 +187,13 @@ impl Def {
 				}
 			}
 
+			if pallet_index.is_none() {
+				if let syn::Item::Type(item) = item {
```

Comment:
This if doesn't make sense? I mean the index can only be set when the item is a type, at least judging based on the code above.

Reply:
> This if doesn't make sense? I mean the index can only be set when the item is a type, at least judging based on the code above.

So, the module marked with #[frame_support::runtime] can have arbitrary items. For example,
```rust
#[frame_support::runtime]
mod runtime {
    use super::*;
    
    pub type System = frame_system;
}
```

In this case, you only want this check for `pub type System = frame_system;` and not for `use super::*;`. 

Basically, this change requires all type aliases within the runtime macro to be annotated with `#[runtime::pallet_index]` while the other items are preserved.